lvsfunc/comparison.py

```diff
@@ -13,6 +17,316 @@
+                if elem:
+                    array_count += 1
+
+        if array_count != self.num_clips:
+            raise ValueError('specified arrangement has an invalid number of clips')
+
+    def _compare(self) -> vs.VideoNode:
+        clips = self._markclips()
+        for row_key in range(len(self.arrangement)):
+            for key, elem in enumerate(self.arrangement[row_key]):
+                self.arrangement[row_key][key] = clips.pop(0) if elem else self.BlankClip
+
+        rows: List[vs.VideoNode] = [core.std.StackHorizontal(row) for row in self.arrangement]
+
+        return core.std.StackVertical(rows)
+
+    def _auto_arrangement(self):
+        dimension = 1 + math.isqrt(self.num_clips - 1)
+        row = [1 for _ in range(dimension)]
+        array = [row.copy() for _ in range(dimension)]
+
+        num_blank_clips = dimension ** 2 - self.num_clips
+
+        if num_blank_clips >= dimension:
+            array.pop(-1)
+            for i in range(1, num_blank_clips - dimension + 1):
+                array[-1][-i] = 0
+        else:
+            for i in range(1, num_blank_clips + 1):
+                array[-1][-i] = 0
+
+        self.arrangement: List[List[Union[int, vs.VideoNode]]] = array
+
+
+class Split(Stack):
+    """
+    Split and unlimited amount of clips into one VideoNode with the same dimensions as the original clips.
+    Handles odd-sized resolutions or resolutions that can't be evenly split by the amount of clips specified.
+
+    The remaining pixel width/height (clip.dimension % number_of_clips) will be always given to the last clip specified.
+    For example, five `104 x 200` clips will result in a `((20 x 200) * 4) + (24 x 200)` horiztonal stack of clips.
+
+    :param direction:       Enum value from :py:class:`lvsfunc.comparison.Direction`.
+                            Determines which dimension will be split (Default: ``comparison.Direction.HORIZONTAL`` )
+    :param label_alignment: Number from 1 to 9, corresponding to the positions of the keys on a numpad.
+                            Determines where to place clip name using ``vapoursynth.core.text.Text`` .
+                            Will default to top-left for horizontal comparison or top-middle for vertical comparison
+                            if not specified (Default: None)
+    :param clips:           Keyword argument `name=clip` pairs for comparison. `name`s will be used to label the clips.
+                            The order the `clips` are specified will be respected in the comparison.
+                            The `width` and `height` for all clips must be the same.
+    """
+
+    def __init__(self, direction: Direction = Direction.HORIZONTAL,
+                 label_alignment: Optional[Literal[1, 2, 3, 4, 5, 6, 7, 8, 9]] = None,
+                 **clips: vs.VideoNode):
+        super().__init__(direction=direction, label_alignment=label_alignment, **clips)
+        self._smart_crop(direction=self.direction)
```

Comment:
This function is only used here, so it should also be defined in the same class and not `Comparer`.

Reply:
Originally was going to code a combination of Tile and Split that was also going to use this, but that's not happening for now so I moved it to Split.